fxdc/parsedata/parsedata.py

```
@@ -106,13 +109,10 @@ def parse(self, preserve_type: bool = True):
                                 obj, key, class_(**newobj.__dict__)
-                            ) if preserve_type else setattr(
-                                obj, key, newobj.__dict__
-                            )
-                        except TypeError:
-                            raise InvalidData(
-                                f"Invalid arguments for class {type_}"
-                            )
+                            ) if preserve_type else setattr(obj, key, newobj.__dict__)
+                        except TypeError as e:
+                            print(e)
+                            raise InvalidData(f"Invalid arguments for class {type_}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)